crates/sui-core/src/safe_client.rs

```diff
@@ -615,13 +615,24 @@ where
                 {
                     // Verify signature.
                     if let Some(signed_proposal) = proposal {
-                        let committee =
+                        let mut committee =
                             self.get_committee(&signed_proposal.auth_signature.epoch)?;
                         signed_proposal.verify(&committee, response.detail.as_ref())?;
                         if signed_proposal.summary.sequence_number > 0 {
                             let cert = prev_cert.as_ref().ok_or_else(|| {
                                 SuiError::from("No checkpoint cert provided along with proposal")
                             })?;
+                            if cert.auth_signature.epoch != signed_proposal.auth_signature.epoch {
+                                // It's possible that the previous checkpoint cert is from the
+                                // previous epoch, and in that case we verify them using different
+                                // committee.
+                                fp_ensure!(
+                                    cert.auth_signature.epoch + 1
```

Comment:
Is it impossible that we end up here with a cert with higher (of equal) epoch number than `signed_proposal.auth_signature.epoch`?

Reply:
The cert must be from the exact previous checkpoint than the proposal. So it's either from the same epoch, or the previous epoch